### ISSUE TITLE
match fails due to Long/Integer discrepency in map keys.

### ISSUE THREAD

**Issue by westbury** [skyscreamer/JSONassert#85]
I tried JSONassert this morning.  The first thing I did was to take something that was output from a test an an org.json.JSONObject structure.  I serialized this using toString(), and pasted that string as the expected value parameter to JSONAssert.assertEquals, passing the JSONObject as the actual value. 
 Unfortunately the comparison failed, so I stepped into the code and found the problem in compareJSONArrayOfJsonObjects.  My structure was using long values but the string was being parsed into Integer values.  The keys of expectedValueMap were Integer whereas the keys of actualValueMap were Long.  #27 partially fixes this issue but fails to fix it in the case where the Integer/Long conflict occurs in a property identified as a unique key.

**Comment by maresja1** [comment]
Yes! It is a bit rubish - tried the library for a first time and came to this the same day. There is no development here either. Wonder how it became recommended library in so many places like Baeldung.

**Comment by ericssy** [comment]
Can you show us a sample test case? I don't quite understand why you can use Integer or Long as keys, since JSON only allows String to be used as keys.
